src/shared-components/segmentedControl/index.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useState, useRef, useEffect } from 'react';
+
+import { SegmentsContainer, SegmentItem, Indicator } from './style';
+import { SegmentedControlProps, SegmentItemType } from './types';
+
+const SegmentedControl: React.FC<SegmentedControlProps> = ({
+  segmentItems,
+  initialActiveId = 1,
+  onClick,
+}) => {
+  const initialActiveItem = segmentItems.find(
+    (item: SegmentItemType) => item.id === initialActiveId,
+  );
+  const initialActiveIndex = initialActiveItem
```

Comment:
Are there any instances in which the `id` and the `index` of the active segment differ by more than one? Could we reduce the state management overhead by just relying on one of them?